src/checkers/inference/solver/backend/BackEndType.java

```diff
@@ -0,0 +1,61 @@
+package checkers.inference.solver.backend;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+import org.checkerframework.javacutil.ErrorReporter;
+
+public class BackEndType {
```

Comment:
It seems we can use a top-level enum class to simplified the code:

```java
public enum BackEndType {
   MAXSAT("maxsat", "fullyqualifiedClassPath"),
   ...;

  public final String simpleName;
  public final String fullyQualifiedName;

  private BackEndType(String simpleName, String fullyQualifiedName) {
   this.simpleName = simpleName;
   this.fullyQualifiedName = fullyQualifiedName;
  }

 public static BackEndType simpleNameToBackEndType(String simpleName) {
    //either for-loop the BackEndType.values array, or pre-cache a static HashSet if we have a lot backends
 }
}
```

As the `simpleName` and `fullyQualifiedName` are final, we can access it directly, and no need to maintain a hashMap. Also, we get rid of the nested class structure.